### ISSUE TITLE
Physical state change not reflected in Zigbee2MQTT for Gledopto GL-SD-003P

### ISSUE THREAD

**Issue by jaco** [Koenkk/zigbee2mqtt#23661]
### What happened?

When changing the state (ON/OFF) of the Gledopto GL-SD-003P device using the physical input, the new state is not reflected in Zigbee2MQTT. The actual state of the device is only updated in Zigbee2MQTT after manually refreshing the state via the web interface.

### What did you expect to happen?

The state change (ON/OFF) via the physical input should be immediately reflected in Zigbee2MQTT without requiring a manual refresh.

### How to reproduce it (minimal and precise)

1. Change the state (ON/OFF) of the device using the physical input (e.g., a wall switch or button)
2. Observe the state of the device in the Zigbee2MQTT web interface. Note that the state does not change to reflect the physical input.
3. Manually refresh the device state via the Zigbee2MQTT web interface.

### Zigbee2MQTT version

1.39.1 

### Adapter firmware version

20220219

### Adapter

SLZB-06

### Setup

Add-on on Home Assistant OS, x86-64

### Debug log

No logs after changing physical state

**Comment by gh0st125** [comment]
I have the exact same issue. Any news on this?

**Comment by ghostrider21991** [comment]
I have the exact same issue. @Koenkk is there any chance u can check this? With the Hue Bridge it works.

**Comment by Koenkk** [comment]
Fixed! After updating press the yellow re-configure button on the frontend.

Changes will be available in the [dev branch](https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html) in a few hours from now.

**Comment by dkkr23** [comment]
I have the exact issue with the model GL-SD-001. @Koenkk  please can you extend the fix to the model as well. Thanks

**Comment by Koenkk** [comment]
@dkkr23 done!

Changes will be available in the [dev branch](https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html) in a few hours from now.

**Comment by dkkr23** [comment]
> [@dkkr23](https://github.com/dkkr23) done!
> 
> Changes will be available in the [dev branch](https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html) in a few hours from now.

Apologies @Koenkk ! The model is GL-SD-001P. I didn’t realise there is a “P” at the end.

**Comment by Koenkk** [comment]
Also fixed for `GL-SD-001P`

Changes will be available in the [dev branch](https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html) in a few hours from now.

**Comment by dkkr23** [comment]
@jaco has this fix resolved the issue for you? I still dont get the state updated with the physical switch.

**Comment by ghostrider21991** [comment]
> [@jaco](https://github.com/jaco) [@gh0st125](https://github.com/gh0st125) [@ghostrider21991](https://github.com/ghostrider21991) has this fix resolved the issue for you? I still dont get the state updated with the physical switch.

Yes, it works. You have to use the EDGE or DEV Version of Zigbee2mqtt. And u also have to klick on the Orange "reconfig button".

**Comment by jaco** [comment]
I have moved inputs to the another controller to prevent messing up the lights…

I will check that next week :)

**Comment by romain3w** [comment]
Hello, I have the problem on the GL-C-003P  (firmware 20951203). I must go on EDGE or DEV Version of Zigbee2mqtt or it's solved now on main version? @Koenkk

**Comment by ghostrider21991** [comment]
It’s solved in the main version. Am 02.02.2026 um 22:12 schrieb romain3w ***@***.***>:﻿romain3w left a comment (Koenkk/zigbee2mqtt#23661)
Hello, I have the problem on the GL-C-003P  (firmware 20951203). I must go on EDGE or DEV Version of Zigbee2mqtt or it's solved now on main version?

**Comment by romain3w** [comment]
I can confirm the issue on GL-C-003P (seen as GL-C-003P)
By default, the device does not report its state correctly. State changes on front push button (on/off) are not reflected unless a manual refresh is triggered.
After manually adding a reporting configuration for the genOnOff cluster (onOff attribute) via the Zigbee2MQTT UI, the device starts working correctly and state updates are reported as expected. Same for genLevelCtrl.

<img width="1574" height="313" alt="Image" src="https://github.com/user-attachments/assets/4ba26df8-6907-4986-8000-b518c7bb8348" />
